### PR TITLE
[#12048] Migrate GetCourseJoinStatusAction

### PR DESCRIPTION
**Pull request by dishenggg**
Part of #12048 

Outline of Solution
- Change `GetCourseJoinStatusAction` to use SQL DB if possible
- Add `getAccountRequestByRegistrationKey` to AccountRequestLogic class
- Add data required for tests to `typicalDataBundle.json`
- Add test cases

### CODE REVIEW

**Comment by weiquu** [comment]
Hi @dishenggg, could you take a look at the failing component test? Can ignore the E2E test. Thanks!

File: src/main/java/teammates/ui/webapi/GetCourseJoinStatusAction.java [hunk 1]
```diff
@@ -38,29 +41,47 @@ public JsonResult execute() {
...
+        if (instructorAttributes == null || isCourseMigrated(instructorAttributes.getCourseId())) {
+            Instructor instructor = sqlLogic.getInstructorByRegistrationKey(regkey);
 
-        return getJoinStatusResult(instructor.isRegistered());
+            if (instructor == null) {
+                throw new EntityNotFoundException("No instructor with given registration key: " + regkey);
+            }
+
+            return getJoinStatusResult(instructor.isRegistered());
+        }
```
Comment (FergusMok) [inline src/main/java/teammates/ui/webapi/GetCourseJoinStatusAction.java:83]: This works, but the logic is abit more complicated and will make removing of the pre-migrated code later on to be more difficult. Will it be better to decouple them and do something similar to lines 60-70? Or even better, you can do something like:

```
if (!isCourseMigrated(...)) {
   // Do datastore things 
} else {
   // Do sql sthings
}
```

File: src/main/java/teammates/ui/webapi/GetCourseJoinStatusAction.java [hunk 2]
```diff
@@ -38,29 +41,47 @@ public JsonResult execute() {
...
-            throw new EntityNotFoundException("No student with given registration key: " + regkey);
+        StudentAttributes studentAttributes = logic.getStudentForRegistrationKey(regkey);
+        if (studentAttributes == null || isCourseMigrated(studentAttributes.getCourse())) {
+            Student student = sqlLogic.getStudentByRegistrationKey(regkey);
+
+            if (student == null) {
+                throw new EntityNotFoundException("No student with given registration key: " + regkey);
+            }
+
+            return getJoinStatusResult(student.isRegistered());
```
Comment (FergusMok) [inline src/main/java/teammates/ui/webapi/GetCourseJoinStatusAction.java:52]: Similar to the other comment

**Review by FergusMok — CHANGES_REQUESTED** [review]
Nice work, just some small nits

File: src/main/java/teammates/ui/webapi/GetCourseJoinStatusAction.java (hunk 1 above)
Comment (dishenggg) [reply src/main/java/teammates/ui/webapi/GetCourseJoinStatusAction.java]: That makes sense! I have made the changes

**Review by cedricongjh — APPROVED** [review]
LGTM, thank you!

**Review by FergusMok — APPROVED** [review]
LGTM!

**Pull request merged**